From 6d5dfaa00364a5baa24f899b9391b04248cac18c Mon Sep 17 00:00:00 2001
From: kunleiz <kunleiz@codeaurora.org>
Date: Tue, 12 Mar 2019 10:27:41 +0800
Subject: [PATCH 2/2] policy_hal: avoid open stereo channel output for voip_rx
 flag

Stereo output is started with setting voip_rx flag. This is not
allowed due to voip_rx profile is not supported.

Add channel mask and sample rate check to ensure voip_rx flag is set
when using correct channel and sample rate.

CRs-Fixed: 2414061
Change-Id: Iae9252915006b7f8b695a1e4680845f2e4e3ac8f
---
 policy_hal/AudioPolicyManager.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/policy_hal/AudioPolicyManager.cpp b/policy_hal/AudioPolicyManager.cpp
index 490eeb4..6a7bdbf 100644
--- a/policy_hal/AudioPolicyManager.cpp
+++ b/policy_hal/AudioPolicyManager.cpp
@@ -1542,7 +1542,10 @@ audio_io_handle_t AudioPolicyManagerCustom::getOutputForDevice(
         }
 #else
     if (stream == AUDIO_STREAM_VOICE_CALL &&
-        audio_is_linear_pcm(config->format)) {
+        audio_is_linear_pcm(config->format) &&
+        (config->channel_mask == 1) &&
+        (config->sample_rate == 8000 || config->sample_rate == 16000 ||
+        config->sample_rate == 32000 || config->sample_rate == 48000)) {
         //check if VoIP output is not opened already
         bool voip_pcm_already_in_use = false;
         for (size_t i = 0; i < mOutputs.size(); i++) {
-- 
2.17.1

